src/t8_cmesh/t8_cmesh_readmshfile.c

```diff
@@ -834,7 +851,13 @@ t8_cmesh_from_msh_file (const char *fileprefix, int partition,
       return NULL;
     }
     /* Check if msh-file version is compatible. */
-    t8_cmesh_check_version_of_msh_file (file);
+    if (t8_cmesh_check_version_of_msh_file (file) != 1) {
+      /* If reading the MeshFormat-number failed or the version is incompatible, close the file */
+      fclose (file);
+      t8_debugf
+        ("The reading process of the msh-file has failed and the file has been closed.\n");
```

Comment:
This return leaves the allocated cmesh dangling and the memory is not freed -> memory leak.

We can fix it by destroying the cmesh before returning:
`t8_cmesh_destroy (&cmesh);`